Add guidance on avoiding using the `maxlength` attribute (#3058)
alphagov/govuk-design-system: src/components/text-input/index.md

Users often need to copy and paste information into a text input, so do not stop them from doing this.

### Avoid restricting the length of a user's input

> Using the `maxlength` attribute can confuse users as it doesn't provide feedback that the text has been truncated, especially in cases where the text has been copied and pasted from elsewhere. This can cause users to accidentally provide incorrect information.

Reviewer: ```suggestion
Using the `maxlength` attribute means there is no feedback to users that their text input is truncated. This is especially true where the text has been copied and pasted from elsewhere. This can cause users to accidentally provide incorrect or incomplete information.
```

> A restrictive maximum length can stop users from formatting information in a way that they're used to. For example, putting spaces in a postcode or commas in a large number.

Reviewer: ```suggestion
A restrictive maximum length can stop users from formatting information in their usual way. For example, putting spaces in a postcode or commas in a large number.
```

Some assistive technologies do not tell users if an input has a `maxlength` set or if the user has passed the limit. Voice control software may insert additional spaces into the input.

> If you must enforce a maximum length for technical reasons, inform the user of the limit in the hint but allow them to provide more information. Only return an error if the value is longer than is allowed after normalisation. For longer values, consider using the [character count component](/components/character-count/) instead.

Reviewer: ```suggestion
If you must enforce a maximum length for technical reasons, inform the user of the limit in the hint, but allow them to provide more information. Only return an error if the value is longer than allowed after normalisation. For longer values, consider using the [character count component](/components/character-count/) instead.
```
